# Getting started with Colab and Python

Let's start with Colab and Python:

1. Import `pandas` library for `DataFrame` manipulation.
2. Download the `Iris` dataset from Kaggle.
3. Load dataset into a `pandas` `DataFrame`.
4. Handle exceptions with `try...except`.
5. Explore the `DataFrame`.
6. Plot data with the `matplotlib` library.
7. Plot data with the `seaborn` library.

# 1. Import the `pandas` library

In [ ]:
import pandas as pd # Python code

If `pandas` is not installed you can install it in the notebook or on the shell (aka terminal):

In [ ]:
!pip install pandas # shell command using the pip Python package manager

I can quickly check if the library was loaded:

In [ ]:
print('pd' in locals())  # check if pd is in the local environment

True


# 2. Download CSV data from Kaggle into `DataFrame`

We will download the `Iris` dataset from Kaggle. We will store the URL in a variable, and then read the CSV data into a `DataFrame`.

In [ ]:
url = 'https://tinyurl.com/iris-data-csv' # save URL as a string


Now we load the dataset with the `read_csv()` method from `pandas` and store it in `df`:

In [ ]:
df = pd.read_csv(url)
print(df)

     sepal_length  sepal_width  petal_length  petal_width    species
0             5.1          3.5           1.4          0.2     setosa
1             4.9          3.0           1.4          0.2     setosa
2             4.7          3.2           1.3          0.2     setosa
3             4.6          3.1           1.5          0.2     setosa
4             5.0          3.6           1.4          0.2     setosa
..            ...          ...           ...          ...        ...
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica

[150 rows x 5 columns]


# 3. Explore the data

To access our `DataFrame` `df`, we now need to treat it like an object instance.

In [ ]:
print(df.head()) # print the first few lines of df

   sepal_length  sepal_width  petal_length  petal_width species
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa


In [ ]:
print(df.tail())  # print the last few lines of df

     sepal_length  sepal_width  petal_length  petal_width    species
145           6.7          3.0           5.2          2.3  virginica
146           6.3          2.5           5.0          1.9  virginica
147           6.5          3.0           5.2          2.0  virginica
148           6.2          3.4           5.4          2.3  virginica
149           5.9          3.0           5.1          1.8  virginica


The `info()` function prints the structure of `df`:

In [ ]:
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   sepal_length  150 non-null    float64
 1   sepal_width   150 non-null    float64
 2   petal_length  150 non-null    float64
 3   petal_width   150 non-null    float64
 4   species       150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB
None


## What is the Iris Dataset?

The Iris dataset is a classic and widely used dataset in machine learning and statistics. It contains 150 samples of iris flowers, with each sample having four features (sepal length, sepal width, petal length, and petal width) and one of three possible species (setosa, versicolor, or virginica).

It is often used for introductory classification tasks due to its simplicity and clear separation of classes.

### Where does it come from?

The Iris dataset was introduced by the British statistician and biologist Ronald Fisher in 1936. While the original source is often cited as his paper, the data is widely available and often presented in educational contexts.

You can find the dataset, for example, on the UCI Machine Learning Repository:
[https://archive.ics.uci.edu/dataset/53/iris](https://archive.ics.uci.edu/dataset/53/iris)

To finish with the exploration, let's get a statistical summary from the dataset:

In [ ]:
print(df.describe())

       sepal_length  sepal_width  petal_length  petal_width
count    150.000000   150.000000    150.000000   150.000000
mean       5.843333     3.057333      3.758000     1.199333
std        0.828066     0.435866      1.765298     0.762238
min        4.300000     2.000000      1.000000     0.100000
25%        5.100000     2.800000      1.600000     0.300000
50%        5.800000     3.000000      4.350000     1.300000
75%        6.400000     3.300000      5.100000     1.800000
max        7.900000     4.400000      6.900000     2.500000
